### PR TITLE
[New] Add numbers.js w/ npm auto-update via single package.json

### PR DESCRIPTION
**Pull request by austin665**
Pull request for issue: #12020 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/numbers/numbers.js
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/numbers
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 79
   - Count of stars: 1436
   - Count of forks: 143
 * NPM download stats (optional):
   - Downloads in the last day: 58
   - Downloads in the last week: 378
   - Downloads in the last month: 2040

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by austin665** [comment]
Oops! Three days already, rebasing and sending. Thanks.

**Comment by austin665** [comment]
@PeterDaveHello Sorry, but I am not able to understand the CI error of it being a file not a directory. Could you please take a look and help. Thank you!

**Comment by sashberd** [comment]
1) The lib does not have git tag, you have to open issue on lib page and ask author to add git tag
2) Files are stored  in src and not in dist library
3) The folder name should be exact as name property at package.json
FYI

**Comment by austin665** [comment]
@sashberd  Thanks. I have created an issue in the numbers.js repo to add the git tags here https://github.com/numbers/numbers.js/issues/155

**Comment by PeterDaveHello** [comment]
Oops, looks weird here, the branch seems to be broken, let me try to fix it!

**Comment by PeterDaveHello** [comment]
Oops, looks weird here, the branch seems to be broken, @sufuf3 can you help fix(squash + rebase) it?

**Comment by austin665** [comment]
Sorry guys, somehow my branch got messed up while rebasing with upsream changes.

**Comment by PeterDaveHello** [comment]
@austin665 hey no worries, are you using GitHub web gui to work with this PR or just on your local machine? We can help you to fix it or fix it for you.

**Comment by austin665** [comment]
@PeterDaveHello Thanks for offering. I was using both, but when the branch got older than three days I had to update from my local since I couldn't find a way to do that from UI. During rebasing and committing it went out of sync and added a lot of commits I guess.

**Comment by PeterDaveHello** [comment]
@austin665 I think the key point is to use rebase instead of merge, rebase your branch with the latest branch, then those commits would gone. Sometimes it'll be even easier to directly using a new branch to fix and overwrite the existing branch instead of fixing the old one, let me know if you need any help.

**Comment by austin665** [comment]
@PeterDaveHello I fixed my indentation but not sure what the other error is referring to, the filename numbers.js is the same as the folder numbers.js that I have created. Please help!

**Comment by sashberd** [comment]
@austin665 If you are using npm auto-update via single package.json you have to remove version from package.json.

**Otherwise, you have to add latest version to you branch.**
e.g. if the latest git tag is 0.6.0, add folder with version name , then copy js to this folder and minify them with peterdavehello minifier tool.

File: ajax/libs/numbers.js/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "author":"Steve Kaliski <sjkaliski@gmail.com> (https://github.com/sjkaliski)",
```
Comment (maruilian11) [inline ajax/libs/numbers.js/package.json]: Could you please put `author` right below the `filename`, we prefer to put `name` and `filename` in the head of the package.json. Thanks!

File: ajax/libs/numbers.js/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "statistics"
+  ],
+  "npmName":"numbers",
+  "npmFileMap":[
+    {
+      "basePath":"src",
+      "files":[
+        "**/*"
+      ]
+    }
```
Comment (maruilian11) [inline ajax/libs/numbers.js/package.json]: Please add an object to fetch `v0.0.2` files.
```
{
  "basePath":"public",
  "files":[
    "numbers*"
  ]
}
```

**Comment by PeterDaveHello** [comment]
@austin665 Let's finish this PR?

**Comment by austin665** [comment]
@PeterDaveHello Yes, please! Is there anything I have to do from my side?

**Comment by PeterDaveHello** [comment]
Looks like you've done the requested changes, I'll review it now!

File: ajax/libs/numbers.js/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "mathematics",
+    "numbers",
+    "statistics"
+  ],
+  "npmName":"numbers",
+  "npmFileMap":[
+    {
+      "basePath":"src",
+      "files":[
+        "**/*"
```
Comment (PeterDaveHello) [inline ajax/libs/numbers.js/package.json]: It's better to use `"**/numbers*"` here, as it's `src` not `dist` folder.

**Comment by austin665** [comment]
@PeterDaveHello Done! Please check!

**Comment by PeterDaveHello** [comment]
Oops, need to update the branch first.

**Comment by PeterDaveHello** [comment]
👍

**Comment by austin665** [comment]
@PeterDaveHello Thanks for having my back. Sorry about so many mistakes.

**Comment by PeterDaveHello** [comment]
@austin665 you don't need to say sorry, we all learn from mistakes :) Feel free to pick more issues, make more mistakes and learn something from it 👍

**Pull request merged**